로그인 페이지 구현

Jo-Myounghee opened the pull request:
## 작업 내용
- 로그인 페이지 UI
![image](https://user-images.githubusercontent.com/48939876/139717734-7f71ba89-ab6e-429b-8d6a-7410580e857f.png)


## 고민한 부분
- 폰트 최적화
- props를 사용한 styled-components의 interface
- `.eslintrc`내 parserOptions의 project 위치 (경로에서 frontend 유무)
- prettier 적용

## 리뷰 포인트
- props를 사용한 styled-components의 interface 

## 관련된 이슈 넘버
#13

Jo-Myounghee committed:
Feat: 로그인 페이지 router 생성

/login에 들어가면 로그인 페이지 나올 수 있게 react router를 사용하여 설정

#13

Jo-Myounghee committed:
Design: 기본 css 적용

reset.css 적용 및 기본 폰트 적용

#13

Jo-Myounghee committed:
Feat: 로그인 페이지 UI 생성

피그마에 있는 로그인 페이지와 동일한 UI 생성

#13

Jo-Myounghee committed:
Fix: eslint설정 경로 수정

frontend 내부에 존재해 앞의 경로 삭제

#13

soosungp33 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
+`;
+
+const Login = () => {
+  return (
+    <>
+      <BackgroundImg imgUrl="/img/sand.jpg">
+        <BackgroundWrap>
+          <Content>
+            <LeftSide>
+              <div>
+                <Logo>
+                  <img src="/img/podo.png" alt="justus logo" height="50px" />
+                  <LogoText>우리끼리</LogoText>
+                </Logo>
+                <LogoInfoText>흩어진 추억을 한 곳에 기록해보세요</LogoInfoText>
+              </div>
+              <img src="/img/btnG_완성형.png" alt="" width="350px" />
+            </LeftSide>
+            <div id="right-side">
+              <Polaroid id="polaroid">
+                <img src="/img/glass.jpg" width="300px" height="300px" alt="" />
+                <img src="/img/life_is_travel.png" height="40px" alt="" />
+              </Polaroid>
+            </div>
+          </Content>
+          <Footer>
+            <FooterContent>
+              <p>Copyright ⓒ Team UNDEFINED. 모든 권리 보유.</p>
+              <div>
+                <span>우리끼리</span>
+                <img src="/img/github_icon.png" height="18px" alt="github icon" />
+                <span className="names">공필상 | 김무성 | 김영한 | 조명희</span>
+              </div>
+            </FooterContent>
+          </Footer>
+        </BackgroundWrap>
+      </BackgroundImg>
+    </>
+  );
+};
```
이 부분 파일로 코드가 나눠지면 더 좋을 것 같습니당!

Jo-Myounghee replied:
여기는 추후 로그인 버튼 로직 추가할 때 나누려고 합니다! 감사합니당

ddb8036631 commented on frontend/src/Router.tsx:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <BrowserRouter>
+      <Switch>
+        <Route path="/login" exact component={() => <Login />}></Route>
```
`component={() => <Login />}` 이거랑 `component={Login}`이거 둘 다 로그인 컴포넌트의 렌더링 부분을 리턴하기에, `<Route path="/login" exact component={Login} />`로 변경해도 좋을 것 같습니다

Jo-Myounghee replied:
추후에 계정 정보 존재 여부에 따라 리턴 값을 나누려고 위와 같이 작성했습니다!
`component={() => (!user ? <Login /> : <Redirect to="/" />)}` 요런 식으루여 !

ddb8036631 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
-function Login() {
-  return;
-}
+const BackgroundImg = styled.div<any>`
```
밑에 styled-components와는 달리 BackgroundImg만 타입을 지정해줬는데, 이유가 있을까요?? any 말고 `HTMLElement`와 같은 타입을 지정하면 오류가 발생했나요??

Jo-Myounghee replied:
어제 밑에서 오류가 나서 anyscript해버렸는데 무성님 코드 보고 수정할 수 있을 것 같아요 감사합니당

ddb8036631 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  background-color: #ffffff90;
```
프로젝트에 사용된 컬러 부분은 `src/components/Style/Color`에 있으니 여기에서 가져다가 쓰면 식별하기 좋을 것 같습니다!!

Jo-Myounghee replied:
넵 감사합니다 !!

ddb8036631 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
+                </Logo>
+                <LogoInfoText>흩어진 추억을 한 곳에 기록해보세요</LogoInfoText>
+              </div>
+              <img src="/img/btnG_완성형.png" alt="" width="350px" />
```
alt가 빠진 것 같아요🥲

Jo-Myounghee replied:
허걱 그러네요 감사합니다

ddb8036631 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
+            </LeftSide>
+            <div id="right-side">
+              <Polaroid id="polaroid">
+                <img src="/img/glass.jpg" width="300px" height="300px" alt="" />
+                <img src="/img/life_is_travel.png" height="40px" alt="" />
```
여기두 alt가 빈 문자열입니다...!

Jo-Myounghee replied:
허거거걱 그러네요 ㅠ 감사합니다

ddb8036631 commented on frontend/src/styles/GlobalStyle.tsx:
```diff
@@ -0,0 +1,54 @@
+import { createGlobalStyle } from 'styled-components';
+
+const GlobalStyle = createGlobalStyle`
+  @import url('https://fonts.googleapis.com/css2?family=Nanum+Gothic:wght@400;700;800&display=swap');
```
[styled-components v5 createGlobalStyle 이슈](https://jungpaeng.tistory.com/70)가 있다고 합니다. 마침 저희가 사용하는 styled-compoents의 버전이 5.1.15이니까 한 번 짚고 넘어가도 좋을 것 같습니다~

Jo-Myounghee replied:
이거는 추후 확인해보고 오겠습니다 !

ddb8036631 reviewed (approved):
코드 잘 봤습니다~ React.FC 타입을 지정해준 컴포넌트가 있고 아닌 컴포넌트가 있는데, 어떤 기준으로 적용하셨나요?

gofeel8 commented on frontend/src/Router.tsx:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <BrowserRouter>
+      <Switch>
+        <Route path="/login" exact component={() => <Login />}></Route>
```
component={() => <Login />}  를  component={Login}   으로 바꿔도 되지 않을까요

혹은

<Route path="/login">
       <Login />
</Route>

으로 바꾸면 어떨까요

Jo-Myounghee replied:
추후에 계정 정보 존재 여부에 따라 리턴 값을 나누려고 위와 같이 작성했습니다!
`component={() => (!user ? <Login /> : <Redirect to="/" />)}` 요런 식으루여 !

Jo-Myounghee replied:
혹은 뒤에 코드가 누락된 듯 한데 알 수 있을까요 ~!

gofeel8 commented on frontend/src/pages/Login/Index.tsx:
```diff
@@ -1,7 +1,132 @@
-function Login() {
-  return;
-}
+const BackgroundImg = styled.div<any>`
```
이게 어제 말한 그 any 구나

Jo-Myounghee replied:
ㅎ_ㅎ넹 오늘 수정하겠습니다

gofeel8 commented on frontend/src/styles/GlobalStyle.tsx:
```diff
@@ -0,0 +1,54 @@
+    font-family: "NanumGothic", sans-serif;
+  }
+`;
+
```
reset 확실하네요😄

gofeel8 reviewed (commented):
LGTM😄

Jo-Myounghee committed:
Improve: 타입 수정

기존에 any 타입에서 imgUrl: string으로 수정

#13

Jo-Myounghee committed:
Improve: styled-component 위치 변경

styled-component의 위치를 파일 하단에 위치시키기로 해서 수정

#13

Jo-Myounghee committed:
Fix: merge 중 충돌 해결

# Conflicts:
#	frontend/package.json
#	frontend/src/App.tsx
#	frontend/yarn.lock
